cap-primitives/src/std/fs/canonicalize.rs

```diff
@@ -19,12 +19,19 @@ pub fn canonicalize(start: &fs::File, path: &Path) -> io::Result<PathBuf> {
 
     // TODO: This is a racy check, though it is useful for testing and fuzzing.
     #[cfg(debug_assertions)]
-    if let Ok(canonical_path) = &result {
+    check_canonicalize(start, path, &result);
+
+    result
+}
+
+#[cfg(debug_assertions)]
+fn check_canonicalize(start: &fs::File, path: &Path, result: &io::Result<PathBuf>) {
+    if let Ok(canonical_path) = result {
         let path_result = open(start, path, OpenOptions::new().read(true));
         let canonical_result = open(start, canonical_path, OpenOptions::new().read(true));
         match (path_result, canonical_result) {
             (Ok(path_file), Ok(canonical_file)) => {
-                assert!(is_same_file(&path_file, &canonical_file)?)
+                assert!(is_same_file(&path_file, &canonical_file).unwrap())
```

Comment:
Actually, I think it would be better from the point of view of bug tracing to panic with a custom message here? I.e., `expect` instead of `unwrap`.